File: 第1回/max.py
import numpy as np
import time
import sys

#配列の最大値を求める関数
def max(x):
    y=x[0]
    for i in x:
        if i>y:
            y=i
    return y

start=time.time()#時間測定開始
l=np.random.rand(int(sys.argv[1]))
print("Max: ",max(l))
fin=time.time()#時間測定終了
print("Time: ",fin-start,"s")

File: 第1回/max.java
public class max {

    public static double max(double[] array) {
        double y = array[0];
   
        for (int i = 0; i < array.length; i++) {
            if (y < array[i]) {
                y = array[i];
            }
        }
        return y;
    }
public static void main(String[] args) { 
    double[] array = new double[Integer.parseInt(args[0]) ];
    for (int i = 0; i < array.length; i++) {
        array[i]=Math.random();
    }

    long start = System.currentTimeMillis();
    double max= max(array); 
    long stop = System.currentTimeMillis();

    System.out.println("Max: "+max);
    System.out.println("Time: " + (double) (stop - start) + "s");
    }
}